src/main/java/no/nav/veilarboppfolging/repository/KafkaProducerRepository.java

```diff
@@ -0,0 +1,21 @@
+package no.nav.veilarboppfolging.repository;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class KafkaProducerRepository {
+    private final JdbcTemplate db;
+
+    @Autowired
+    public KafkaProducerRepository(JdbcTemplate db) {
+        this.db = db;
+    }
+
+    public Integer getNumberOfUnprocessedMessages() {
+        return db.queryForObject("SELECT COUNT(*) FROM KAFKA_PRODUCER_RECORD WHERE CREATED_AT < SYSDATE - INTERVAL '30' MINUTE",
```

Comment:
Maybe we can return the oldest CREATED_AT date? or null or now() if there are no messages, and then grafana can trigger the alert if the date is too old?

Reply:
or maybe return now() - "oldest CREATED_AT" to get how old the oldest message is?

Reply:
Aha, yep, that can be useful as well